lib/cfg-grammar.y

```diff
@@ -814,10 +814,23 @@ template_content_inner
           CHECK_ERROR_GERROR(log_template_set_type_hint(last_template, $1, &error), @1, error, "Error setting the template type-hint \"%s\"", $1);
           free($1);
         }
+        | LL_NUMBER
```

Comment:
I think we could even support null as a sole token, however before adding these, I'd want to implement support for further literals as we extend the syntax for filter expressions.

For the record, this is what I came up with:

```
diff --git a/lib/cfg-grammar.y b/lib/cfg-grammar.y
index 8a80adb6c..3b1b03272 100644
--- a/lib/cfg-grammar.y
+++ b/lib/cfg-grammar.y
@@ -814,6 +814,14 @@ template_content_inner
           CHECK_ERROR_GERROR(log_template_set_type_hint(last_template, $1, &error), @1, error, "Error setting the template type-hint \"%s\"", $1);
           free($1);
         }
+        | LL_IDENTIFIER
+        {
+          GError *error = NULL;
+
+          CHECK_ERROR_GERROR(log_template_compile(last_template, "", &error), @0, error, "Error compiling empty template");
+          CHECK_ERROR_GERROR(log_template_set_type_hint(last_template, $1, &error), @1, error, "Error setting the template type-hint \"%s\"", $1);
+          free($1);
+        }
         | LL_NUMBER
         {
           gchar decimal[32];
```

This could work also by adding an opening/closing bracket, but I am afraid using LL_IDENTIFIER here will generate conflicts down the line. I'd probably need to add "null" as a keyword, which might cause issues with existing configurations where I've seen "null" being used unquoted as a destination name.

All in all, I like this suggestion, but let get this merged and add these down the line.